reporting-cli/src/download-helpers.js

```diff
@@ -0,0 +1,341 @@
+    exit(1);
+  }
+
+  await page.waitForTimeout(2000);
+  await page.click('button[data-test-subj="confirm"]');
+  await page.waitForTimeout(25000);
+  await overridePage.goto(url, { waitUntil: 'networkidle0' });
+  await overridePage.waitForTimeout(5000);
+
+  // Check if tenant was selected successfully.
+  if ((await overridePage.$('button[data-test-subj="confirm"]')) !== null) {
+    spinner.fail('Invalid tenant');
+    exit(1);
+  }
+  await page.goto(url, { waitUntil: 'networkidle0' });
+};
+
+const samlAuthentication = async (page, url, username, password, tenant) => {
+  await page.goto(url, { waitUntil: 'networkidle0' });
+  await new Promise(resolve => setTimeout(resolve, 10000));
+  let refUrl;
+  await getUrl(url).then((value) => {
+    refUrl = value;
+  });
+  await page.type('[name="identifier"]', username);
+  await page.type('[name="credentials.passcode"]', password);
+  await page.click('[value="Sign in"]')
+  await page.waitForTimeout(30000);
+  try {
+    if (tenant === 'global' || tenant === 'private') {
+      await page.click('label[for=' + tenant + ']');
+    } else {
+      await page.click('label[for="custom"]');
+      await page.click('button[data-test-subj="comboBoxToggleListButton"]');
+      await page.type('input[data-test-subj="comboBoxSearchInput"]', tenant);
+    }
+  }
+  catch (err) {
+    spinner.fail('Invalid username or password');
+    exit(1);
+  }
+  await page.waitForTimeout(2000);
+  await page.click('button[data-test-subj="confirm"]');
+  await page.waitForTimeout(25000);
+  await page.click(`a[href='${refUrl}']`);
+}
+
+const cognitoAuthentication = async (page, overridePage, url, username, password, tenant) => {
+  await page.goto(url, { waitUntil: 'networkidle0' });
+  await new Promise(resolve => setTimeout(resolve, 10000));
+  await page.type('[name="username"]', username);
+  await page.type('[name="password"]', password);
+  await page.click('[name="signInSubmitButton"]');
+  await page.waitForTimeout(30000);
+  try {
+    if (tenant === 'global' || tenant === 'private') {
+      await page.click('label[for=' + tenant + ']');
+    } else {
+      await page.click('label[for="custom"]');
+      await page.click('button[data-test-subj="comboBoxToggleListButton"]');
+      await page.type('input[data-test-subj="comboBoxSearchInput"]', tenant);
+    }
+  }
+  catch (err) {
+    spinner.fail('Invalid username or password');
+    exit(1);
+  }
+  await page.waitForTimeout(2000);
+  await page.click('button[data-test-subj="confirm"]');
+  await page.waitForTimeout(25000);
+  await overridePage.goto(url, { waitUntil: 'networkidle0' });
+  await overridePage.waitForTimeout(5000);
+
+  // Check if tenant was selected successfully.
+  if ((await overridePage.$('button[data-test-subj="confirm"]')) !== null) {
+    spinner.fail('Invalid tenant');
+    exit(1);
+  }
+  await page.goto(url, { waitUntil: 'networkidle0' });
+}
+
+export const readStreamToFile = async (
+  stream,
+  fileName,
```

Comment:
nit. no need to export?

Also should fileName be customizable?

Reply:
removed export. fileName is customizable. User can use `-n` or `--filename` to customize report's name.